src/controller/Controller.hs

```diff
@@ -11,5 +14,29 @@ input :: Event -> GameState -> IO GameState
 input _ = return
 
 -- This function takes an event, a certain keyboard input, and a current gamestate, and returns the new gamestate.
--- keyboardInputHandler :: Event -> GameState -> GameState
--- keyboardInputHandler (EventKey (SpecialKey KeyUp))
\ No newline at end of file
+keyboardInputHandler :: Event -> GameState -> GameState
+keyboardInputHandler (EventKey pressedButton _ _ _) gameState  = case phase gameState of
+  -- If the gamephase is 'playing', the following input is considered:
+  Playing -> case pressedButton of
+    SpecialKey KeyUp -> gameState {player =  movePlayer (player gameState) verticalMovementStep} -- Move the player upwards.
```

Comment:
Deze regels lezen misschien net iets makkelijker als je `player` al uit `gameState` haalt aan het begin van de functie bij de input parameters. Je kunt iets vergelijkbaars doen als ik wat op veel plekken heb gedaan in de GameState file: `gameState@GameState{player}`. De @ zorgt ervoor dat je de hele gamestate aan kunt roepen met `gameState`, maar je ook toegang hebt tot de uitgepakte versie met het `player` veld (je kunt natuurlijk ook andere velden toevoegen). Op die manier hoef je niet de hele tijd `player gameState` te doen om bij de player te komen.